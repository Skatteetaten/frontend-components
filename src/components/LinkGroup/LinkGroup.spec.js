import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { LinkGroup } from '../index';

function oppsettMount() {
  return mount(<LinkGroup links={[{ text: 'Dette er en link', path: '#' }]} />);
}

describe('Link komponent', () => {
  it('matcher snapshot', () => {
    const wrapper = oppsettMount();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
