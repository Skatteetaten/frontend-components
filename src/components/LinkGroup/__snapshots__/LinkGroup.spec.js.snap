// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link komponent matcher snapshot 1`] = `
<LinkGroup
  links={
    Array [
      Object {
        "path": "#",
        "text": "Dette er en link",
      },
    ]
  }
>
  <ul
    className="arrowLinkList-37"
  >
    <li
      className="arrowLink-36"
      key="0"
    >
      <Link
        icon="ArrowForward"
        linkGroup={true}
        path="#"
        placement="before"
        text="Dette er en link"
      >
        <span
          className="linkContainer-39"
        >
          <Icon
            className="icon-38"
            iconName="ArrowForward"
            role="presentation"
          >
            <StyledIconBase
              className="icon-38"
              iconName="ArrowForward"
              role="presentation"
            >
              <IconBase
                className="icon-38"
                iconName="ArrowForward"
                role="presentation"
                styles={[Function]}
                theme={
                  Object {
                    "disableGlobalClassNames": false,
                    "effects": Object {
                      "elevation16": "0 6.4px 14.4px 0 rgba(0, 0, 0, 0.132), 0 1.2px 3.6px 0 rgba(0, 0, 0, 0.108)",
                      "elevation4": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
                      "elevation64": "0 25.6px 57.6px 0 rgba(0, 0, 0, 0.22), 0 4.8px 14.4px 0 rgba(0, 0, 0, 0.18)",
                      "elevation8": "0 3.2px 7.2px 0 rgba(0, 0, 0, 0.132), 0 0.6px 1.8px 0 rgba(0, 0, 0, 0.108)",
                      "roundedCorner2": "2px",
                    },
                    "fonts": Object {
                      "large": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "18px",
                        "fontWeight": 300,
                      },
                      "medium": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "16px",
                        "fontWeight": 400,
                      },
                      "mediumPlus": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "17px",
                        "fontWeight": 400,
                      },
                      "mega": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "72px",
                        "fontWeight": 100,
                      },
                      "small": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "14px",
                        "fontWeight": 400,
                      },
                      "smallPlus": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "15px",
                        "fontWeight": 400,
                      },
                      "superLarge": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "42px",
                        "fontWeight": 100,
                      },
                      "tiny": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "10px",
                        "fontWeight": 600,
                      },
                      "xLarge": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "22px",
                        "fontWeight": 100,
                      },
                      "xLargePlus": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
                        "fontSize": "24px",
                        "fontWeight": 600,
                      },
                      "xSmall": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "12px",
                        "fontWeight": 400,
                      },
                      "xxLarge": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Helvetica Neue', Helvetica , Arial, sans-serif",
                        "fontSize": "28px",
                        "fontWeight": 100,
                      },
                      "xxLargePlus": Object {
                        "MozOsxFontSmoothing": "grayscale",
                        "WebkitFontSmoothing": "antialiased",
                        "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
                        "fontSize": "32px",
                        "fontWeight": 600,
                      },
                    },
                    "isInverted": false,
                    "palette": Object {
                      "accent": "#1362ae",
                      "black": "rgba(11, 11, 11, 0.95)",
                      "blackTranslucent40": "rgba(0,0,0,.4)",
                      "blue": "#0078d4",
                      "blueDark": "#002050",
                      "blueLight": "#00bcf2",
                      "blueMid": "#00188f",
                      "bodyBackground": "#ffffff",
                      "bodyText": "rgba(0, 0, 0, 0.95)",
                      "disabledBackground": "#f4f4f4",
                      "disabledText": "#c8c8c8",
                      "green": "#107c10",
                      "greenDark": "#004b1c",
                      "greenLight": "#bad80a",
                      "magenta": "#b4009e",
                      "magentaDark": "#5c005c",
                      "magentaLight": "#e3008c",
                      "neutralDark": "rgba(21, 21, 21, 0.95)",
                      "neutralLight": "#eaeaea",
                      "neutralLighter": "#f4f4f4",
                      "neutralLighterAlt": "#f8f8f8",
                      "neutralPrimary": "rgba(0, 0, 0, 0.95)",
                      "neutralPrimaryAlt": "rgba(47, 47, 47, 0.95)",
                      "neutralQuaternary": "#d0d0d0",
                      "neutralQuaternaryAlt": "#dadada",
                      "neutralSecondary": "rgba(55, 55, 55, 0.95)",
                      "neutralSecondaryAlt": "#8a8886",
                      "neutralTertiary": "rgba(89, 89, 89, 0.95)",
                      "neutralTertiaryAlt": "#c8c8c8",
                      "orange": "#d83b01",
                      "orangeLight": "#ea4300",
                      "orangeLighter": "#ff8c00",
                      "primaryBackground": "#ffffff",
                      "primaryText": "rgba(0, 0, 0, 0.95)",
                      "purple": "#5c2d91",
                      "purpleDark": "#32145a",
                      "purpleLight": "#b4a0ff",
                      "red": "#e81123",
                      "redDark": "#a4262c",
                      "skeColor": Object {
                        "beige": "#f9ede2",
                        "black": "#000000",
                        "blackAlt": "#1d1d1d",
                        "blue": "#1362ae",
                        "brown": "#e7b78a",
                        "burgundy": "#6f2c3f",
                        "burgundyLight": "#b07c83",
                        "darkBeige": "#f3dbc4",
                        "darkBlue": "#093e61",
                        "darkGrey": "#666666",
                        "error": "#a82e39",
                        "green": "#91d6ac",
                        "grey": "#999999",
                        "internal": "#6F2C3F",
                        "internalLight": "#ECE6E8",
                        "lightBeige": "#fcf6f0",
                        "lightBlue": "#cde1f9",
                        "lightBrown": "#ecc7a3",
                        "lightGreen": "#e3f5ea",
                        "lightGrey": "#bbbbbb",
                        "lightPink": "#f7d1d7",
                        "mediumBlue": "#8accff",
                        "neutralGrey": "#f4f4f4",
                        "pink": "#df4661",
                        "transparent": "transparent",
                        "white": "#ffffff",
                        "whiteGrey": "#e8e8e8",
                      },
                      "teal": "#008272",
                      "tealDark": "#004b50",
                      "tealLight": "#00b294",
                      "themeDark": "#0c3f6f",
                      "themeDarkAlt": "#11599c",
                      "themeDarker": "#0a365f",
                      "themeLight": "#82bcf1",
                      "themeLighter": "#cce3f9",
                      "themeLighterAlt": "#ebf4fd",
                      "themePrimary": "#1362ae",
                      "themeSecondary": "#146aba",
                      "themeTertiary": "#288be7",
                      "white": "#ffffff",
                      "whiteTranslucent40": "rgba(255,255,255,.4)",
                      "yellow": "#ffb900",
                      "yellowDark": "#d29200",
                      "yellowLight": "#fff100",
                    },
                    "rtl": undefined,
                    "semanticColors": Object {
                      "accentButtonBackground": "#1362ae",
                      "accentButtonText": "#ffffff",
                      "actionLink": "rgba(0, 0, 0, 0.95)",
                      "actionLinkHovered": "rgba(21, 21, 21, 0.95)",
                      "blockingBackground": "#FDE7E9",
                      "blockingIcon": "#FDE7E9",
                      "bodyBackground": "#ffffff",
                      "bodyBackgroundChecked": "#eaeaea",
                      "bodyBackgroundHovered": "#f4f4f4",
                      "bodyDivider": "#eaeaea",
                      "bodyFrameBackground": "#ffffff",
                      "bodyFrameDivider": "#eaeaea",
                      "bodyStandoutBackground": "#f8f8f8",
                      "bodySubtext": "rgba(55, 55, 55, 0.95)",
                      "bodyText": "rgba(0, 0, 0, 0.95)",
                      "bodyTextChecked": "rgba(11, 11, 11, 0.95)",
                      "buttonBackground": "#ffffff",
                      "buttonBackgroundChecked": "#c8c8c8",
                      "buttonBackgroundCheckedHovered": "#eaeaea",
                      "buttonBackgroundDisabled": "#f4f4f4",
                      "buttonBackgroundHovered": "#f4f4f4",
                      "buttonBackgroundPressed": "#eaeaea",
                      "buttonBorder": "#8a8886",
                      "buttonBorderDisabled": "#f4f4f4",
                      "buttonText": "rgba(0, 0, 0, 0.95)",
                      "buttonTextChecked": "rgba(21, 21, 21, 0.95)",
                      "buttonTextCheckedHovered": "rgba(11, 11, 11, 0.95)",
                      "buttonTextDisabled": "rgba(89, 89, 89, 0.95)",
                      "buttonTextHovered": "rgba(21, 21, 21, 0.95)",
                      "buttonTextPressed": "rgba(21, 21, 21, 0.95)",
                      "defaultStateBackground": "#f8f8f8",
                      "disabledBackground": "#f4f4f4",
                      "disabledBodySubtext": "#c8c8c8",
                      "disabledBodyText": "rgba(89, 89, 89, 0.95)",
                      "disabledBorder": "#c8c8c8",
                      "disabledSubtext": "#d0d0d0",
                      "disabledText": "rgba(89, 89, 89, 0.95)",
                      "errorBackground": "#FDE7E9",
                      "errorIcon": "#A80000",
                      "errorText": "#a4262c",
                      "focusBorder": "rgba(55, 55, 55, 0.95)",
                      "infoBackground": "#f3f2f1",
                      "infoIcon": "#605e5c",
                      "inputBackground": "#ffffff",
                      "inputBackgroundChecked": "#1362ae",
                      "inputBackgroundCheckedHovered": "#0c3f6f",
                      "inputBorder": "rgba(55, 55, 55, 0.95)",
                      "inputBorderHovered": "rgba(0, 0, 0, 0.95)",
                      "inputFocusBorderAlt": "#1362ae",
                      "inputForegroundChecked": "#ffffff",
                      "inputIcon": "#1362ae",
                      "inputIconDisabled": "rgba(89, 89, 89, 0.95)",
                      "inputIconHovered": "#0c3f6f",
                      "inputPlaceholderBackgroundChecked": "#cce3f9",
                      "inputPlaceholderText": "rgba(55, 55, 55, 0.95)",
                      "inputText": "rgba(0, 0, 0, 0.95)",
                      "inputTextHovered": "rgba(21, 21, 21, 0.95)",
                      "link": "#1362ae",
                      "linkHovered": "#0a365f",
                      "listBackground": "#ffffff",
                      "listHeaderBackgroundHovered": "#f4f4f4",
                      "listHeaderBackgroundPressed": "#eaeaea",
                      "listItemBackgroundChecked": "#eaeaea",
                      "listItemBackgroundCheckedHovered": "#dadada",
                      "listItemBackgroundHovered": "#f4f4f4",
                      "listText": "rgba(0, 0, 0, 0.95)",
                      "listTextColor": "rgba(0, 0, 0, 0.95)",
                      "menuBackground": "#ffffff",
                      "menuDivider": "#c8c8c8",
                      "menuHeader": "#1362ae",
                      "menuIcon": "#1362ae",
                      "menuItemBackgroundChecked": "#eaeaea",
                      "menuItemBackgroundHovered": "#f4f4f4",
                      "menuItemBackgroundPressed": "#eaeaea",
                      "menuItemText": "rgba(0, 0, 0, 0.95)",
                      "menuItemTextHovered": "rgba(21, 21, 21, 0.95)",
                      "messageLink": "#005A9E",
                      "messageLinkHovered": "#004578",
                      "messageText": "#323130",
                      "primaryButtonBackground": "#1362ae",
                      "primaryButtonBackgroundDisabled": "#f4f4f4",
                      "primaryButtonBackgroundHovered": "#11599c",
                      "primaryButtonBackgroundPressed": "#0c3f6f",
                      "primaryButtonBorder": "transparent",
                      "primaryButtonText": "#ffffff",
                      "primaryButtonTextDisabled": "#d0d0d0",
                      "primaryButtonTextHovered": "#ffffff",
                      "primaryButtonTextPressed": "#ffffff",
                      "severeWarningBackground": "#FED9CC",
                      "severeWarningIcon": "#D83B01",
                      "smallInputBorder": "rgba(55, 55, 55, 0.95)",
                      "successBackground": "#DFF6DD",
                      "successIcon": "#107C10",
                      "successText": "#107C10",
                      "variantBorder": "#eaeaea",
                      "variantBorderHovered": "rgba(89, 89, 89, 0.95)",
                      "warningBackground": "#FFF4CE",
                      "warningHighlight": "#ffb900",
                      "warningIcon": "#797775",
                      "warningText": "#323130",
                    },
                    "spacing": Object {
                      "l1": "20px",
                      "l2": "32px",
                      "m": "16px",
                      "s1": "8px",
                      "s2": "4px",
                    },
                  }
                }
              >
                <i
                  aria-hidden={true}
                  className="icon-43"
                  data-icon-name="ArrowForward"
                  role="presentation"
                >
                  
                </i>
              </IconBase>
            </StyledIconBase>
          </Icon>
          <a
            className="iconLink-41"
            href="#"
          >
            Dette er en link
          </a>
        </span>
      </Link>
    </li>
  </ul>
</LinkGroup>
`;
